### PR TITLE
Add ERC: dApp Permission Framework

### PR DESCRIPTION
**Pull request by jqphu**
Adding initial draft to get ERC number then creating a post on ethereum magicians.

### CODE REVIEW

File: ERCS/eip-xxxx.md [hunk 1]
```diff
@@ -0,0 +1,266 @@
+---
```
Comment (abcoathup) [inline ERCS/eip-xxxx.md:1]:
```suggestion
---
eip: 7601
```
Assigning next sequential EIP/ERC/RIP number.
Numbering changed to sequential from 7500 and is no longer the PR number.

Please also update the filename.

File: ERCS/eip-xxxx.md [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+title: dApp Permission Framework (DPF)
+description: Declarative permissions to minimize impact when a web dApp is compromised.
+author: Justin Phu (@jqphu)
+discussions-to: <URL>
+status: Draft
+type: Standards Track
+category: Interface
+created: 2024-02-12
+requires: [EIP712](../EIPS/eip-712.md]
```
Comment (SamWilsn) [inline ERCS/eip-xxxx.md:10]:
```suggestion
requires: 712
```

File: ERCS/eip-xxxx.md [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+---
+
+## Abstract
+
+There have been many instances of official web dApps being compromised. There are a plethora of ways this can happen such as a supply chain attack, a malicious code update, DNS compromises (e.g. DNS credential compromise or BGP attack).
+
+In this proposal, we propose a mechanism for dApps to declare what interactions they allow which will vastly reduce the efficacy of these attacks and in some situations prevent them entirely.
+
+![[../assets/eip-xxxx/dApp_permission_framework_high_level.png]]
```
Comment (SamWilsn) [inline ERCS/eip-xxxx.md:19]: Your abstract could stand to have a bit more technical detail. This section should give the reader a brief/high-level (but still technical) overview of your proposal.

File: ERCS/eip-xxxx.md [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+
+## Motivation
+
+Phishing attacks in crypto are incredibly devastating as transactions are final and there is no recourse. There has been a recent surge in phishing scams whereby the official websites are compromised.
+
+Examples of these are
+- Premint - https://cryptoslate.com/nft-platform-premint-users-lose-over-400k-nfts-to-hack/
+- Ledger Connect Supply Chain - https://techcrunch.com/2023/12/14/supply-chain-attack-targeting-ledger-crypto-wallet-leaves-users-hacked/
+- Celar Bridge: https://www.certik.com/resources/blog/1NHvPnvZ8EUjVVs4KZ4L8h-bgp-hijacking-how-hackers-circumvent-internet-routing-security-to-tear-the
```
Comment (SamWilsn) [inline ERCS/eip-xxxx.md:28]: Please remove external links.

File: ERCS/eip-xxxx.md [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+Phishing scams compromise websites trick users into thinking they're doing a normal interaction but replace it with a malicious one.
+
+When phishing scams are able to compromise a website they fire off many different scam types to steal as many assets as possible. These interactions can be as simple as requesting to send ETH to another wallet or as complex as leveraging the OpenSea protocol to trick users to selling their NFTs for nothing.
+
+This proposal will attempt to sandbox dApps to make it much more difficult for an attacker to extract funds from users when a dApp is compromised.
+
+Example:
+We will look at `opensea.io` and evaluate what would happen if the website was compromised.
+
+Under typical operation `opensea.io` interacts with OpenSea Contracts (ignoring approvals for simplicity).
```
Comment (SamWilsn) [inline ERCS/eip-xxxx.md:39]: Please remove mentions of specific products.

File: assets/eip-xxxx/DPF_compromised.png
Comment (SamWilsn) [inline assets/eip-xxxx/DPF_compromised.png:1]: We have a slight preference for SVG images. If you can't provide them, PNG is fine too.

File: ERCS/eip-xxxx.md [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+Note: this means, teams should update their DNS Records to add more permissions **before** updating the dApp itself.
+
+### How it all fits together
+
+Let's walk through the full example for `opensea.io`.
+
+#### Developers
+
+> [!todo]
+> Fill this with exact values.
```
Comment (SamWilsn) [inline ERCS/eip-xxxx.md:182]: If you use HTML-style comments (`<!-- TODO -->`), the linter will make sure you replace them before going into review.

**Review by SamWilsn — COMMENTED** [review]
I've only given this a quick look. Please fix the issues reported by `eipw`.

**Pull request closed without merging**